### PR TITLE
fix: error extractor to handle nested JSON responses

### PR DESCRIPTION
**Pull request by koladilip**
# Description

This PR enhances the error extractor in the enterprise reporting module to properly handle nested JSON responses. The fix addresses cases where error messages are embedded within nested JSON structures, particularly when the outer JSON contains a 'response' field with a JSON string that includes error information.

## Changes Made

- **Enhanced Error Extractor Logic**: Added a fallback mechanism in the  function to handle direct message keys in parsed JSON
- **Improved Nested JSON Parsing**: Better handling of nested JSON response structures for more accurate error message extraction
- **Comprehensive Test Coverage**: Added extensive test cases covering various nested JSON scenarios, edge cases, and real-world examples
- **Backward Compatibility**: Maintained all existing functionality while adding new capabilities

## Linear Ticket

Resolves https://linear.app/rudderstack/issue/INT-3995/fix-error-extractor-not-handling-nested-json-response-correctly

## Security

- [x] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

## Testing

- [x] All existing tests pass
- [x] New comprehensive test coverage added for nested JSON scenarios
- [x] Edge cases and error conditions tested
- [x] No user data included in test cases (all data sanitized)

## Files Changed

-  - Enhanced error extraction logic
-  - Added comprehensive test coverage

### CODE REVIEW

File: enterprise/reporting/error_extractor.go [hunk 1]
```diff
@@ -33,7 +33,7 @@ var (
 	spaceRegex       = regexp.MustCompile(`\s+`)
 	whitespacesRegex = regexp.MustCompile("[ \t\n\r]*") // used in checking if string is a valid json to remove extra-spaces
 
-	defaultErrorMessageKeys = []string{"message", "description", "detail", errorKey, "title", "error_message"}
+	defaultErrorMessageKeys = []string{"error_message", "message", "description", "detail", errorKey, "title"}
```
Comment (ItsSudip) [inline enterprise/reporting/error_extractor.go:37]: Can you tell me why we have to change the order here?

Comment (koladilip) [reply enterprise/reporting/error_extractor.go:37]: to capture the reddit error the, check test case example

**Review by ItsSudip — APPROVED** [review]
